src/Modules/UngradedModules.jl

```diff
@@ -5280,23 +5280,45 @@ end
 
 compose(h::ModuleFPHom, g::ModuleFPHom) = h*g
 
-# TODO: We need to also make all of the code below sensitive to base changes!
--(h::ModuleFPHom) = hom(domain(h), codomain(h), [-h(x) for x in gens(domain(h))])
-function -(h::ModuleFPHom, g::ModuleFPHom)
+-(h::ModuleFPHom{D, C, Nothing}) where {D, C} = hom(domain(h), codomain(h), [-h(x) for x in gens(domain(h))])
+-(h::ModuleFPHom{D, C, T}) where {D, C, T} = hom(domain(h), codomain(h), [-h(x) for x in gens(domain(h))], base_ring_map(h))
+
+function -(h::ModuleFPHom{D, C, T}, g::ModuleFPHom{D, C, T}) where {D, C, T}
+  @assert domain(h) === domain(g)
```

Comment:
I think `@req` which is coming from Hecke is the more appropriate macro for input checks
because the @assert macro may be ignored by the compiler in the future. (See its documentation:
```
  │ Warning
  │
  │  An assert might be disabled at various optimization levels. Assert
  │  should therefore only be used as a debugging tool and not used for
  │  authentication verification (e.g., verifying passwords), nor
  │  should side effects needed for the function to work correctly be
  │  used inside of asserts.
```

Reply:
Ok, good to know. But I'm not using it, usually.